### ISSUE TITLE
Error message can be made better by removing duplicate entries

### ISSUE THREAD

**Issue by sowmyav27** [rancher/rke#1975]
**RKE version:**
v1.1.0-rc16

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
AWS

**cluster.yml file:**
```
nodes:
  - address: <ip-1>
    internal_address: <p-ip-1>
    user: ubuntu
    role: [controlplane,worker,etcd]
    ssh_key_path: <>
  - address: <ip-2>
    internal_address: <p-ip-2>
    user: ubuntu
    role: [controlplane,worker,etcd]
    ssh_key_path: <>
  - address: <ip-3>
    internal_address: <p-ip-3>
    user: ubuntu
    role: [controlplane,worker,etcd]
    ssh_key_path: <>

services:
  etcd:
    snapshot: true
    creation: 6h
    retention: 24h
```

**Steps to Reproduce:**
Deploy a cluster expired with certs (using rke cli binary with certs expiry in 15 minutes)
2. Take a snapshot
3. Wait for the cluster certs to expire
4. restore from snapshot (taken in step 2) using rke cli 1.1.0-rc16 gives me error - 
```
./rke etcd snapshot-restore --name test_certs --config rancher-cluster.yml
WARN[0000] This is not an officially supported version (v1.1.0-rc16) of RKE. Please download the latest official release at https://github.com/rancher/rke/releases/latest 
INFO[0000] Running RKE version: v1.1.0-rc16             
INFO[0000] Restoring etcd snapshot test_certs           
FATA[0000] [etcd] Failed to restore etcd snapshot: invalid certs: Certificate [kube-node] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-etcd-172-31-4-104] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-apiserver] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-scheduler] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-etcd-172-31-0-167] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-controller-manager] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-proxy] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-etcd-172-31-6-140] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-ca] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-apiserver-proxy-client] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-apiserver-proxy-client] failed verification: x509: certificate has expired or is not yet valid 
```
5. Rotate certs - `./rke cert rotate --config rancher-cluster.yml` (rotates all certs except kube-ca)
6. Error seen when trying to restore to snapshot (taken in step 2) - 
```
./rke etcd snapshot-restore --name test_certs --config rancher-cluster.yml
WARN[0000] This is not an officially supported version (v1.1.0-rc16) of RKE. Please download the latest official release at https://github.com/rancher/rke/releases/latest 
INFO[0000] Running RKE version: v1.1.0-rc16             
INFO[0000] Restoring etcd snapshot test_certs           
FATA[0000] [etcd] Failed to restore etcd snapshot: invalid certs: Certificate [kube-ca] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-ca] failed verification: x509: certificate has expired or is not yet valid 
```

**Results:**
The error message in Step 6 has duplicate entries which can be removed.

**Related to** - https://github.com/rancher/rke/issues/1928

**Comment by mrajashree** [comment]
Fix available with v1.1.0-rc17

**Comment by sowmyav27** [comment]
**Verified on rke `v1.1.0-rc17`**

- Because this issue is fixed -- https://github.com/rancher/rke/issues/1974, user will NOT be able to do a cert rotate like `./rke cert rotate --config rancher-cluster.yml` when the kube-ca certs are expired.

**Steps:**
- Bring up a cluster with cert expiry - 5 mins
- Take a snapshot
- wait for the cluster to expire
- Restore to the snapshot using  rke `v1.1.0-rc17`.

```
./rke etcd snapshot-restore --name test --config rancher-cluster.yml 
WARN[0000] This is not an officially supported version (v1.1.0-rc17) of RKE. Please download the latest official release at https://github.com/rancher/rke/releases/latest 
INFO[0000] Running RKE version: v1.1.0-rc17             
INFO[0000] Restoring etcd snapshot test                 
FATA[0000] [etcd] Failed to restore etcd snapshot: invalid certs: Certificate [kube-controller-manager] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-ca] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-apiserver] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-node] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-etcd-172-31-9-11] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-etcd-172-31-2-77] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-apiserver-proxy-client] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-proxy] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-etcd-172-31-10-30] failed verification: x509: certificate has expired or is not yet valid: Certificate [kube-scheduler] failed verification: x509: certificate has expired or is not yet valid 
```
- No duplicates seen in this error message.
